app/src/main/java/io/kuenzler/whatsappwebtogo/WebviewActivity.java

```diff
@@ -277,6 +286,14 @@ public void onUnhandledKeyEvent(WebView view, KeyEvent event) {
         }
 
         mWebView.getSettings().setUserAgentString(USER_AGENT);
+        
+          ///Checking and requesting permission at runtime
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            if (checkPermission()) {
+            } else {
+                requestPermission(); // Code for permission
+            }
+        }
```

Comment:
Until now the app uses proper runtime permissions. The permission should only be asked for if necessary